src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs

```diff
@@ -0,0 +1,217 @@
+        =========================================================================*/
+        public ThreadPriority Priority
+        {
+            [SecuritySafeCritical]  // auto-generated
+            get { return (ThreadPriority)GetPriorityNative(); }
+            [SecuritySafeCritical]  // auto-generated
+            [HostProtection(SelfAffectingThreading = true)]
+            set { SetPriorityNative((int)value); }
+        }
+
+        [SecurityCritical]  // auto-generated
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private extern int GetPriorityNative();
+
+        [SecurityCritical]  // auto-generated
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private extern void SetPriorityNative(int priority);
+
+        /*=========================================================================
+        ** Return the thread state as a consistent set of bits.  This is more
+        ** general then IsAlive or IsBackground.
+        =========================================================================*/
+        public ThreadState ThreadState
+        {
+            [SecuritySafeCritical]  // auto-generated
+            get { return (ThreadState)GetThreadStateNative(); }
+        }
+
+        [SecurityCritical]  // auto-generated
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        private extern int GetThreadStateNative();
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public ApartmentState GetApartmentState()
+        {
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            return (ApartmentState)GetApartmentStateNative();
+#else // !FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            Contract.Assert(false); // the Thread class in CoreFX should have handled this case
+            return ApartmentState.MTA;
+#endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        }
+
+        /*=========================================================================
+        ** An unstarted thread can be marked to indicate that it will host a
+        ** single-threaded or multi-threaded apartment.
+        =========================================================================*/
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [HostProtection(Synchronization = true, SelfAffectingThreading = true)]
+        public bool TrySetApartmentState(ApartmentState state)
+        {
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            return SetApartmentStateHelper(state, false);
+#else // !FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            Contract.Assert(false); // the Thread class in CoreFX should have handled this case
+            return false;
+#endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        }
+
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        [SecurityCritical]  // auto-generated
+        protected bool SetApartmentStateHelper(ApartmentState state, bool fireMDAOnMismatch)
```

Comment:
protected members are visible outside the assembly - not something you meant. These should be internal.